### PR TITLE
[DifferenceKit] Upgrade to Difference Kit 1.0.0

### PR DESCRIPTION
**Pull request by anayini**
DifferenceKit finally had its 1.0.0 release, so this commit
upgrades to it.

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/plangrid/ReactiveLists/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by jessesquires** [comment]
need to update the podspec too

**Comment by anayini** [comment]
@jessesquires more than what I already changed?

**Comment by jessesquires** [comment]
@anayini Yes. 

https://github.com/plangrid/ReactiveLists/blob/master/ReactiveLists.podspec

File: ReactiveLists.podspec [hunk 1]
```diff
@@ -16,5 +16,5 @@ Pod::Spec.new do |s|
   s.swift_version = '4.2'
   s.requires_arc = true
 
-  s.dependency 'DifferenceKit', '~> 0.8.0'
+  s.dependency 'DifferenceKit', '~> '1.0.0'
```
Comment (jessesquires) [inline ReactiveLists.podspec]: ah sorry. here it is. missed it in the diff earlier 😅

**Comment by PlanBot-iOS** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">There is no milestone for this PR. Please add one.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**